examples/vector_fe/vector_fe_ex7/vector_fe_ex7.C

```diff
@@ -0,0 +1,551 @@
+  // less than the dimensionality of the element.
+  QGauss qface(dim - 1, FIFTH);
+
+  vector_fe_face->attach_quadrature_rule(&qface);
+  enriched_scalar_fe_face->attach_quadrature_rule(&qface);
+  lambda_fe_face->attach_quadrature_rule(&qface);
+
+  const auto & JxW = vector_fe->get_JxW();
+  const auto & q_point = vector_fe->get_xyz();
+  const auto & vector_phi = vector_fe->get_phi();
+  const auto & scalar_phi = scalar_fe->get_phi();
+  const auto & div_vector_phi = vector_fe->get_div_phi();
+  const auto & vector_phi_face = vector_fe_face->get_phi();
+  const auto & enriched_scalar_phi_face = enriched_scalar_fe_face->get_phi();
+  const auto & lambda_phi_face = lambda_fe_face->get_phi();
+  const auto & JxW_face = vector_fe_face->get_JxW();
+  const auto & qface_point = vector_fe_face->get_xyz();
+  const auto & normals = vector_fe_face->get_normals();
+
+  // We follow the notation of Cockburn
+  // LM matrix and RHS
+  DenseMatrix<Number> E_libmesh;
+  DenseVector<Number> H_libmesh;
+  MatrixXd E;
+  MatrixXd H;
+  // Auxiliary matrices and RHS
+  MatrixXd A, Ainv, B, Bt, C, Ct, Sinv;
+  VectorXd G, F, Hg, Hf;
+  // element matrix for boundary LM dofs
+  MatrixXd L;
+
+  // Lambda eigen vector for constructing vector and scalar solutions
+  VectorXd Lambda;
+  // The lambda solution at the quadrature points
+  std::vector<Number> lambda_qps;
+
+  /// Data structures for computing the enriched scalar solution
+  DenseMatrix<Number> K_enriched_scalar;
+  DenseVector<Number> F_enriched_scalar, U_enriched_scalar;
+
+  auto zero_mat = [](auto & mat)
+  {
+    for (const auto i : make_range(mat.rows()))
+      for (const auto j : make_range(mat.cols()))
+        mat(i, j) = 0;
+  };
+  auto zero_vec = [](auto & vec)
+  {
+    for (const auto i : make_range(vec.size()))
+      vec(i) = 0;
+  };
+
+  std::vector<dof_id_type> vector_dof_indices;
+  std::vector<dof_id_type> scalar_dof_indices;
+  std::vector<dof_id_type> enriched_scalar_dof_indices;
+  std::vector<dof_id_type> lambda_dof_indices;
+  std::vector<Number> lambda_solution_std_vec;
+
+  // The global system matrix
+  SparseMatrix<Number> & matrix = lambda_system.get_system_matrix();
+
+  for (const auto & elem : mesh.active_local_element_ptr_range())
+  {
+    dof_map.dof_indices(elem, vector_dof_indices, system.variable_number("u"));
+    dof_map.dof_indices(elem, scalar_dof_indices, system.variable_number("p"));
+    lambda_dof_map.dof_indices(elem, lambda_dof_indices, lambda_system.variable_number("lambda"));
+
+    const auto vector_n_dofs = vector_dof_indices.size();
+    const auto scalar_n_dofs = scalar_dof_indices.size();
+    const auto lambda_n_dofs = lambda_dof_indices.size();
+
+    vector_fe->reinit(elem);
+    scalar_fe->reinit(elem);
+
+    libmesh_assert_equal_to(vector_n_dofs, vector_phi.size());
+    libmesh_assert_equal_to(scalar_n_dofs, scalar_phi.size());
+
+    // vector equation
+    A.resize(vector_n_dofs, vector_n_dofs);
```

Comment:
I think there's
```
A = MatrixXd::Zero(vector_n_dofs, vector_n_dofs);
```
if you want to drop those lambdas.

Reply:
Actually, I think even more appropriate would be:
```
A.setZero(vector_n_dofs, vector_n_dofs);
```

Reply:
Oh nice

Reply:
Sorry to belabor the minor point, but I believe `setZero(m,n)` handles the resizing, so you don't need both `resize()` and `setZero()`. The following test code works as expected for me anyway:
```
  Eigen::MatrixXd A;
  A.setZero(3, 10);
  libMesh::out << "A.rows() = " << A.rows() << std::endl;
  libMesh::out << "A.cols() = " << A.cols() << std::endl;
  libMesh::out << A << std::endl;
```

Reply:
I feel so assaulted 😄